Update GUI labels to reflect multiple rate providers

cd2357 opened the pull request:
Update the Bisq GUI labels to better indicate the source of the exchange rate data.

The areas changed are:

1) labels in the top-right box showing the exchange rate
2) label in the About view

**1) Top right box / The shown text for fiat and alts:**

![Screenshot from 2020-08-09 13-09-40](https://user-images.githubusercontent.com/15956136/89730773-3760aa00-da42-11ea-9147-13903f31e02d.png)


**1) Top right box / The tooltip for fiat and alts:**

![Screenshot from 2020-08-09 13-58-55](https://user-images.githubusercontent.com/15956136/89731607-ad680f80-da48-11ea-9b75-55bfc6e34f4b.png)

![Screenshot from 2020-08-09 13-59-11](https://user-images.githubusercontent.com/15956136/89731609-af31d300-da48-11ea-81c1-005e965f6673.png)

**2) About view / Source of market prices under Data Providers**

![Screenshot from 2020-08-09 15-32-50](https://user-images.githubusercontent.com/15956136/89733521-494c4800-da56-11ea-9532-84c22b1f9900.png)


---

Addresses bisq-network/projects#35

Related to #4315

cd2357 committed:
Update top-right bar references to price source

Update the displayed text, as well as the tooltip, of the price box in
the top right bar. It now indicates that the price data is provided by
Bisq pricenodes (for for fiat, as well as for alts).

cd2357 committed:
Correctly map lastRequest timestamp

The lastRequest timestamp is changed to show the last request to a
pricenode.

The previous approach of using the "last provider request timestamp"
does not make sense in the new setup. Each currency rate is based on
rates from several providers, each with their own "request timestamps".
In addition, the pricenode returns the timestamp each rate was
calculated. On top of that comes the timestamp when the Bisq node
queries the pricenode.

Since what is most relevant for the Bisq node is the "freshness" of a
specific rate, the timestamp most indicative of that is the moment when
the pricenode is queried.

cd2357 committed:
Remove unused methods

Removed getters for the BA and CMC timestamps, both of which are not
used anymore.

cd2357 committed:
Rename method

Rename method to remove reference to the BA provider.

cd2357 committed:
Rename timestamp field

Rename timestamp field which implied it represents an epoch value in
seconds, but the way it was used to build a Date object showed that it
actually expected a millis value.

cd2357 commented:
@wiz this is the label and tooltip change, in a separate PR.

wiz commented on core/src/main/java/bisq/core/provider/price/PriceFeedService.java:
```diff
@@ -400,7 +384,12 @@ public void onSuccess(@Nullable Tuple2<Map<String, Long>, Map<String, MarketPric
+                    // Each currency rate has a different timestamp, depending on when
+                    // the pricenode aggregate rate was calculated
+                    // However, the request timestamp is when the pricenode was queried
+                    epochInMillisAtLastRequest = System.currentTimeMillis();
```
Hold on a second, this isn't just changing labels, you're actually changing code here. It looks like you're replacing the timestamp of the actual price data, with the timestamp Bisq updated it from the Pricenode, which is incorrect. This behavior needs to be preserved, for example if a pricenode has an issue and doesn't get data updates, the local Bisq node needs to be able to detect that it's old data and switch to a different Pricenode with recently updated data.

cd2357 replied:
Since there is no more BA provider, the `btcAverageTs` field is basically useless.

It is however used here to populate the `epochInMillisAtLastRequest` timestamp -- which is used in the UI to show when the fiat rates were retrieved.

Before, when using BA: this made sense, cause BA was the "single source of truth" for all fiat rates, and Poloniex was the source of alt rates.

Now, when using multiple providers: basically the pricenode itself is the "provider", because it aggregates rates from multiple sources for both fiat and alts.

So, the closest "timestamp" that shows the last time the "provider" (pricenode) was polled, is the one in the code snippet above.

Alternative would be to extract and index every timestamp per exchange rate, then selectively show the right one depending on what currency pair the user selected in the UI. That seems like overkill for what this change tries to achieve. Especially since those "per exchange rate" timestamps are very likely the same, cause they all show the point in time when the avg rate was calculated, and they're all calculated when the pricenode is polled (the avg is freshly calculated based on newest data).

So I would say, this is the simplest and most pragmatic approach (`epochInMillisAtLastRequest` = `timestamp_when_pricenode_last_successfully_queried`).

wiz replied:
The timestamp is not for when the data was updated from the pricenode, that's irrelevant. The timestamp is for the price itself, i.e. when the price was obtained from the provider, or in the case of a weighted average aka index, the time when the index was last updated. Can you please use the `timestampSec` field for each asset's index?

cd2357 replied:
Done. This field now just stores the "pricenode query timestamp", but the `timestampSec` of each asset is shown in the tooltip (see PR screenshots).

cd2357 committed:
Update labels to indicate Bisq Price Index

Update top-right status box and tooltip labels to indicate the price is
based on the Bisq Price Index.

cd2357 committed:
Update top-right tooltip timestamp

Update timestamp shown in top-right tooltip, to indicate the point in
time when that specific exchange rate was retrieved (from an Exchange,
if only one exchange supported for that currency) or when it was
calculated (by the pricenode, based on inputs from multiple exchanges).

cd2357 committed:
Update exchange rate info in About view

Update field describing the source of the shown exchange rates,
indicating that the Bisq Price Index is used.

wiz reviewed (dismissed):
ACK ef5b80440cda2eee7de2f5c155eedfaf042cb3dd
looks great!
<img width="216" alt="Screen Shot 2020-08-10 at 1 32 14" src="https://user-images.githubusercontent.com/232186/89737187-03b96a80-daaa-11ea-9e42-5fa2fb186638.png">
<img width="522" alt="Screen Shot 2020-08-10 at 1 34 23" src="https://user-images.githubusercontent.com/232186/89737182-0025e380-daaa-11ea-8fa1-2912a590ef95.png">
<img width="569" alt="Screen Shot 2020-08-10 at 1 33 07" src="https://user-images.githubusercontent.com/232186/89737188-03b96a80-daaa-11ea-98cd-8f8cfb8e3096.png">

wiz commented on core/src/main/resources/i18n/displayStrings.properties:
```diff
@@ -1166,7 +1165,6 @@ setting.about.providers=Data providers
 setting.about.apisWithFee=Bisq uses 3rd party APIs for Fiat and Altcoin market prices as well as for mining fee estimation.
 setting.about.apis=Bisq uses 3rd party APIs for Fiat and Altcoin market prices.
```
```suggestion
+setting.about.apisWithFee=Bisq uses Bisq Price Indices for Fiat and Altcoin market prices, and Bisq Mempool Nodes for mining fee estimation.
+setting.about.apis=Bisq uses Bisq Price Indices for Fiat and Altcoin market prices.
```

wiz replied:
Trusted Third Parties are security holes :wink:

cd2357 committed:
Update core/src/main/resources/i18n/displayStrings.properties

Co-authored-by: wiz <j@wiz.biz>

sqrrm reviewed (commented):
How do we handle roll back of the Bisq index in case of issues? Might be better to wait with merging this PR.

cd2357 commented:
> How do we handle roll back of the Bisq index in case of issues? Might be better to wait with merging this PR.

Yes, I'd also suggest to wait with this PR. At least until the pricenode PR runs for a while and you're confident it doesn't need a rollback.

wiz commented:
![just-merge-it](https://user-images.githubusercontent.com/232186/90707778-47467c80-e2d3-11ea-8d33-02e6cb21ef96.jpg)

sqrrm reviewed (approved):
utACK
